Cpp_test/lib/file-handler.cpp

```diff
@@ -0,0 +1,60 @@
+#include "file-handler.h"
+#include <ctime>
+#include <iostream>
+
+FileHandler::~FileHandler()
+{
+	if (fs.is_open())
+		fs.close();
+}
+
+unsigned long FileHandler::get_file_size()
+{
+	if (this->mode == FileMode::WRITE)
+	{
+		this->close_file();
+	}
+	return boost::filesystem::file_size(this->file_name);
```

Comment:
As requested, boost library should not be used.

Reply:
Can you suggest to me the other way do it, except boost?
Because I also use boost in test case to test transfer file

Reply:
You can take a look at: https://linux.die.net/man/2/stat64

Reply:
`stat64` doesn't need a file/fstream opened to operate.

Reply:
I mean I already changed it, by using fstream at commit 719a2cb

Reply:
Ok. But for your information, seekg fails for a file greater than 2GB

https://stackoverflow.com/questions/24385254/at-what-size-of-file-seekg-or-tellg-fails

Reply:
**stat** is good option here

Reply:
I recommend using stat64 instead of stat.